### PR TITLE
Add exclude option to phpmd

### PR DESCRIPTION
**Pull request by jyggen**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documented?   | yes
| Fixed tickets | -

Use phpmd's `--exclude` option to exclude directories from the task.

### CODE REVIEW

File: src/GrumPHP/Task/PhpMd.php [hunk 1]
```diff
@@ -64,6 +66,8 @@ public function run(ContextInterface $context)
         $arguments->addCommaSeparatedFiles($files);
         $arguments->add('text');
         $arguments->addOptionalCommaSeparatedArgument('%s', $config['ruleset']);
+        $arguments->add('--exclude');
```
Comment (veewee) [inline src/GrumPHP/Task/PhpMd.php]: Shouldn't the `--exclude` option be optional as well?

File: doc/tasks/phpmd.md [hunk 2]
```diff
@@ -8,21 +8,28 @@ It lives under the `phpmd` namespace and has following configurable parameters:
 parameters:
     tasks:
         phpmd:
+            exclude: ~
```
Comment (veewee) [inline doc/tasks/phpmd.md]: The default value is []

File: src/GrumPHP/Task/PhpMd.php (hunk 1 above)
Comment (jyggen) [inline src/GrumPHP/Task/PhpMd.php]: You're absolutely right. Fixed! :)

File: doc/tasks/phpmd.md (hunk 2 above)
Comment (jyggen) [inline doc/tasks/phpmd.md]: Yep. Fixed :+1:

File: src/GrumPHP/Task/PhpMd.php [hunk 3]
```diff
@@ -64,6 +66,7 @@ public function run(ContextInterface $context)
         $arguments->addCommaSeparatedFiles($files);
         $arguments->add('text');
         $arguments->addOptionalCommaSeparatedArgument('%s', $config['ruleset']);
+        $arguments->addOptionalCommaSeparatedArgument('--exclude %s', $config['exclude']);
```
Comment (veewee) [inline src/GrumPHP/Task/PhpMd.php]: This one will be quoted as one argument:

```
'phpmd' '--exclude file1,file2'
```

Shouldn't this be:

```
'phpmd' '--exclude' 'file1,file2'
```

Comment (jyggen) [inline src/GrumPHP/Task/PhpMd.php]: Fixed! Not my day today :P

**Comment by veewee** [comment]
Thanks!

**Pull request merged**